components/org.wso2.carbon.aws/src/main/java/org/wso2/carbon/aws/user/store/mgt/AWSUserStoreManager.java

```diff
@@ -0,0 +1,1858 @@
+
+        RoleContext roleContext = createRoleContext(roleName);
+        roleName = roleContext.getRoleName();
+        if (log.isDebugEnabled()) {
+            log.debug("Searching for role " + roleName);
+        }
+        boolean isExistingRole = checkExistenceOfUserOrRole(pathToRoles, roleName);
+        if (log.isDebugEnabled()) {
+            log.debug("Role: " + roleName + " is exists in user store");
+        }
+        return isExistingRole;
+    }
+
+    /**
+     * Adds the user to the user store.
+     *
+     * @param userName              of new user.
+     * @param credential            of new user.
+     * @param roleList              of new user.
+     * @param claims                user claim values.
+     * @param profileName           user profile name.
+     * @param requirePasswordChange Status to change password.
+     * @throws UserStoreException If error occurred.
+     */
+    @Override
+    public void doAddUser(String userName, Object credential, String[] roleList, Map<String, String> claims,
+                          String profileName, boolean requirePasswordChange) throws UserStoreException {
+
+        if (!checkUserNameValid(userName)) {
+            String errorMsg = realmConfig.getUserStoreProperty(AWSConstants.PROPERTY_USER_NAME_ERROR_MSG);
+            if (errorMsg != null) {
+                throw new UserStoreException(errorMsg);
+            }
+            throw new UserStoreException(String.format("User name not valid. It must be a non null string " +
+                    "with following format: %s.", realmConfig.getUserStoreProperty(
+                    UserCoreConstants.RealmConfig.PROPERTY_USER_NAME_JAVA_REG_EX)));
+        }
+        if (!checkUserPasswordValid(credential)) {
+            String errorMsg = realmConfig.getUserStoreProperty(AWSConstants.PROPERTY_PASS_ERROR_MSG);
+            if (errorMsg != null) {
+                throw new UserStoreException(errorMsg);
+            }
+            throw new UserStoreException(String.format("Credential not valid. It must be a non null string " +
+                    "with following format: %s", realmConfig.getUserStoreProperty(
+                    UserCoreConstants.RealmConfig.PROPERTY_JAVA_REG_EX)));
+        }
+        boolean isExisting = doCheckExistingUser(userName);
+        if (isExisting) {
+            throw new UserStoreException(String.format("User name : %s exists in the system. Please pick another " +
+                    "user name", userName));
+        }
+
+        Map<String, String> map = new HashMap<>();
+        byte[] passwordToStore = UserCoreUtil.getPasswordToStore(credential, passwordHashMethod, false);
+
+        map.put(userNameAttribute, userName);
+        map.put(passwordAttribute, new String(passwordToStore));
+        boolean hasRoles = roleList != null && roleList.length > 0;
+        if (membershipType.equals(AWSConstants.ATTRIBUTE) && hasRoles) {
```

Comment:
To avoid possible null pointer swap >> (AWSConstants.ATTRIBUTE).equals(membershipType)